editor.planx.uk/src/@planx/components/List/Public/Fields.tsx

```diff
@@ -3,135 +3,177 @@ import FormControl from "@mui/material/FormControl";
+          type="number"
+          value={formik.values.userData[activeIndex][data.fn]}
+          onChange={formik.handleChange}
+          errorMessage={getIn(
+            formik.errors,
+            `userData[${activeIndex}][${data.fn}]`,
+          )}
+          inputProps={{
+            "aria-describedby": [
+              data.description ? DESCRIPTION_TEXT : "",
+              getIn(formik.errors, `userData[${activeIndex}][${data.fn}]`)
+                ? `${ERROR_MESSAGE}-${id}`
+                : "",
+            ]
+              .filter(Boolean)
+              .join(" "),
+          }}
           id={id}
-          title={data.text}
-          onChange={() => console.log("change radio")}
         />
-      ))}
-    </RadioGroup>
-    {/* </ErrorWrapper> */}
-  </FormControl>
-);
+        {data.units && <InputRowLabel>{data.units}</InputRowLabel>}
+      </Box>
+    </InputLabel>
+  );
+};
+
+export const RadioFieldInput: React.FC<Props<QuestionField>> = ({
+  id,
+  data,
+}) => {
+  const { formik, activeIndex } = useListContext();
+
+  return (
+    <FormControl sx={{ width: "100%" }} component="fieldset">
+      <FormLabel
+        component="legend"
+        id={`radio-buttons-group-label-${id}`}
+        sx={(theme) => ({
+          color: theme.palette.text.primary,
+          "&.Mui-focused": {
+            color: theme.palette.text.primary,
+          },
+        })}
+      >
+        {data.title}
+      </FormLabel>
+      <ErrorWrapper
+        id={`${id}-error`}
+        error={getIn(formik.errors, `userData[${activeIndex}][${data.fn}]`)}
+      >
+        <RadioGroup
+          aria-labelledby={`radio-buttons-group-label-${id}`}
+          name={`userData[${activeIndex}][${[data.fn]}]`}
+          sx={{ p: 1 }}
+          value={formik.values.userData[activeIndex][data.fn]}
+        >
+          {data.options.map(({ id, data }) => (
+            <BasicRadio
+              key={id}
+              id={data.text}
+              title={data.text}
+              onChange={formik.handleChange}
+            />
+          ))}
+        </RadioGroup>
+      </ErrorWrapper>
+    </FormControl>
+  );
+};
 
 export const SelectFieldInput: React.FC<Props<QuestionField>> = ({
   id,
   data,
   required,
-}) => (
-  <InputLabel label={data.title} id={`select-label-${id}`}>
-    <SelectInput
-      bordered
-      required={required}
-      title={data.title}
-      labelId={`select-label-${id}`}
-    >
-      {data.options.map((option) => (
-        <MenuItem key={option.id} value={option.data.text}>
-          {option.data.text}
-        </MenuItem>
-      ))}
-    </SelectInput>
-  </InputLabel>
-);
+}) => {
+  const { formik, activeIndex } = useListContext();
+
+  return (
+    <InputLabel label={data.title} id={`select-label-${id}`}>
+      <ErrorWrapper
+        id={`${id}-error`}
+        error={getIn(formik.errors, `userData[${activeIndex}][${data.fn}]`)}
+      >
+        <SelectInput
```

Comment:
This is our only public-facing `Select` component - we use a MUI `Autocomplete` in FileUploadAndLabel.

Certainly there may be come questions around a11y and usability here which we may want to spend some time on. Initially the a11y unit test failed here, but following [the MUI docs](https://mui.com/material-ui/react-select/#accessibility) resolved this.